google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -199,15 +206,24 @@ private Publisher(Builder builder) throws IOException {
             StatusCode.Code.UNAVAILABLE)
         .setRetrySettings(retrySettingsBuilder.build())
         .setBatchingSettings(BatchingSettings.newBuilder().setIsEnabled(false).build());
-    this.publisherStub = GrpcPublisherStub.create(stubSettings.build());
+    this.publisherStub =
+        this.enableRESTJsonTransport
+            ? HttpJsonPublisherStub.create(stubSettings.build())
+            : GrpcPublisherStub.create(stubSettings.build());
     backgroundResourceList.add(publisherStub);
     backgroundResources = new BackgroundResourceAggregation(backgroundResourceList);
     shutdown = new AtomicBoolean(false);
     messagesWaiter = new Waiter();
-    this.publishContext = GrpcCallContext.createDefault();
+    this.publishContext =
+        this.enableRESTJsonTransport
+            ? HttpJsonCallContext.createDefault()
+            : GrpcCallContext.createDefault();
     this.publishContextWithCompression =
-        GrpcCallContext.createDefault()
-            .withCallOptions(CallOptions.DEFAULT.withCompression(GZIP_COMPRESSION));
+        this.enableRESTJsonTransport
```

Comment:
Note: If the user attempts to enable compression on the JSON transport, should the SDK throw an exception or silently ignore the same as the HttpJsonCallContext does not support compression ?